src/manager.cpp

```diff
@@ -277,51 +290,99 @@ sdbusplus::async::task<void>
     co_return;
 }
 
+sdbusplus::async::task<bool>
+    // NOLINTNEXTLINE
+    Manager::retrySync(const config::DataSyncConfig& cfg, fs::path srcPath,
+                       size_t retryCount, size_t rsyncExitCode)
+{
+    const fs::path currentSrcPath = srcPath.empty() ? cfg._path : srcPath;
+
+    if (retryCount++ < cfg._retry->_retryAttempts)
+    {
+        lg2::debug(
+            "Retry {RETRY_ATTEMPT}/{MAX_ATTEMPTS} for [{SRC_PATH}] after {RETRY_INTERVAL}s "
+            "due to rsync error code {RSYNC_ERR}",
+            "RETRY_ATTEMPT", retryCount, "MAX_ATTEMPTS",
+            cfg._retry->_retryAttempts, "SRC_PATH", currentSrcPath,
+            "RETRY_INTERVAL", cfg._retry->_retryIntervalInSec.count(),
+            "RSYNC_ERR", rsyncExitCode);
+
+        co_await sleep_for(_ctx, std::chrono::seconds(
+                                     cfg._retry->_retryIntervalInSec.count()));
+
+        co_return co_await syncData(cfg, std::move(srcPath), retryCount);
+    }
+
+    // NOTE: The following line is commented out as part of a temporary
+    // workaround. We are forcing Full Sync to succeed even if data syncing
+    // fails. This change should be reverted once proper error handling is
+    // implemented.
+    // setSyncEventsHealth(SyncEventsHealth::Critical);
+
+    lg2::error(
+        "Sync failed after {MAX_ATTEMPTS} retries (exit code: {ERROR_CODE}) for [{SRC_PATH}]",
+        "MAX_ATTEMPTS", cfg._retry->_retryAttempts, "ERROR_CODE", rsyncExitCode,
+        "SRC_PATH", currentSrcPath);
+    co_return false;
+}
+
 sdbusplus::async::task<bool>
     // NOLINTNEXTLINE
     Manager::syncData(const config::DataSyncConfig& dataSyncCfg,
-                      fs::path srcPath)
+                      fs::path srcPath, size_t retryCount)
 {
+    constexpr size_t rsyncVanishedExitCode = 24;
+    const fs::path currentSrcPath = srcPath.empty() ? dataSyncCfg._path
+                                                    : srcPath;
     std::string syncCmd{};
-    getRsyncCmd(RsyncMode::Sync, dataSyncCfg, srcPath.string(), syncCmd);
+    const bool haveIncList = getRsyncCmd(RsyncMode::Sync, dataSyncCfg,
+                                         srcPath.string(), syncCmd);
+
+    if (haveIncList)
+    {
+        // IncludeList with no existing entries no-op success
```

Comment:
It seems sync is skipped if none of the configured include paths exist, otherwise it fails during the full sync.

But, what happens if a configured path does not exist? Should we handle that case as well?